File: backend/src/Host/RecallSmith.Api/Models/Deck.cs
// backend/src/Host/RecallSmith.Api/Models/Deck.cs
namespace RecallSmith.Api.Models
{
    /// <summary>
    /// Authoring 侧的 Deck 模型，对应数据库表 decks。
    /// </summary>
    public class Deck
    {
        public int Id { get; set; }

        /// <summary>
        /// 稳定标识，对应 decks.slug（例如 "js-core-starter"）
        /// </summary>
        public string Slug { get; set; } = null!;

        public string Title { get; set; } = null!;

        public string Author { get; set; } = null!;

        public string? Description { get; set; }

        /// <summary>
        /// 语言，如 "en-US"
        /// </summary>
        public string Locale { get; set; } = "en-US";

        /// <summary>
        /// 1 = Starter, 2 = Paid
        /// </summary>
        public short DeckType { get; set; }

        public int IsDeleted { get; set; }

        public int Version { get; set; }

        public long CreatedAt { get; set; }

        public long UpdatedAt { get; set; }
    }
}

File: frontend/src/types/deck.ts
// src/types/deck.ts

export type DeckTier = 'free' | 'premium';
export type DeckAvailability = 'live' | 'coming' | 'retired';

export interface Deck {
  id: number;
  slug: string;
  title: string;
  author: string;
  description?: string | null;
  locale: string;
  deckType: number;

  canRead?: boolean;
  canWrite?: boolean;

  version: number;
  createdAt: string;
  updatedAt: string;
  isDeleted?: number;

  // ✅ mobile/manifest fields (DB-backed)
  tier?: DeckTier | null; // null => infer by deckType
  availability?: DeckAvailability | null;
  eta?: string | null;
  manifestOrder?: number | null;
  totalCards?: number | null;
  previewCards?: number | null;
  retiredAtMs?: number | null;
  // The build the manifest serves, from decks.live_build_id. Lets editors, who
  // never fetch the admin manifest, still tell a published deck from a draft.
  liveBuildId?: string | null;

  // legacy/optional (keep if you already use)
  contentVersion?: string | null;
  isFreeStarter?: boolean | null;
  freeCardCount?: number | null;
}